jsonschema/_utils.py

```diff
@@ -13,7 +14,9 @@ class URIDict(MutableMapping):
     """
 
     def normalize(self, uri):
-        return urlsplit(uri).geturl()
+        result = urisplit(uri)
```

Comment:
Hmm, if we're concerned about the performance of these methods, we should probably cache this. I suggest passing in the cache as a special keyword argument (py2 doesn't support kw-only args), or by using a factory which returns a new type of URIDict with a particular cache configuration. Thoughts?

Reply:
I wouldn't worry about performance until/unless it's observed to be slow (in which case something like that sounds reasonable).

Reply:
I mention performance because after publishing the other PR (now closed), I noticed that the use of explicit push_ / pop_ scope methods was as a result of loop unrolling due to performance (three years ago).